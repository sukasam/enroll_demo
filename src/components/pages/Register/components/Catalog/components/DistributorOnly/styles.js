import { css } from "@emotion/react";

export default css`
    margin: 20px 0;
    h3 {
        margin: 0 0 10px 0;
    }
    .continue-button {
        font-size: 18px;
        color: white;
        font-weight: 600;
        background-color: #5a8fc3;
        border: none;
        border-radius: 30px;
        padding: 10px;
        cursor: pointer;
        -webkit-transition: background-color 0.3s ease;
        transition: background-color 0.3s ease;
        max-width: 230px;
        width: 100%;
        justify-content: center;
        display: flex;
    }
`;
