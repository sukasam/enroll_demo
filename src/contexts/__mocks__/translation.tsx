import { Alpha2 } from "Constants/countryConfig/enums.js";

export const useTranslations = jest.fn().mockReturnValue({
    country: "US" as Alpha2,
    language: "en",
    setCountry: jest.fn(),
    setLanguage: jest.fn(),
    fetchTranslations: jest.fn(),
    translations: {}
});

export default useTranslations;
