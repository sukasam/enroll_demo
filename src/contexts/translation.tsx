import { debounce } from "@mui/material";
import { translationsSvc } from "Hydra/translationsSvc";
import { sendEvent } from "Services/googleAnalytics";
import { getCookie, setCookie } from "cookies-next";
import {
    ReactNode,
    createContext,
    useCallback,
    useContext,
    useEffect,
    useMemo,
    useState
} from "react";

export interface TranslationContextType {
    translations: Record<string, string>;
    basicTranslations: Record<string, string>;
    country: string;
    language: string;
    setTranslations: (translations: Record<string, string>) => void;
    setBasicTranslations: (basicTranslations: Record<string, string>) => void;
    setCountry: (country: string) => void;
    setLanguage: (language: string) => void;
    fetchTranslations: (country: string, language: string) => void;
}

const defaultState: TranslationContextType = {
    translations: {},
    basicTranslations: {},
    country: getCookie("country") || "US",
    language: getCookie("language") || "en",
    setTranslations: () => null,
    setBasicTranslations: () => null,
    setCountry: () => null,
    setLanguage: () => null,
    fetchTranslations: () => null
};

function getInitialState(): TranslationContextType {
    if (typeof window !== "undefined") {
        const cachedData = sessionStorage.getItem("translationContext");
        if (cachedData) {
            try {
                return JSON.parse(cachedData);
            } catch (error) {
                console.error(
                    "Failed to parse translation context data from sessionStorage:",
                    error
                );
            }
        }
    }
    return defaultState;
}

const TranslationContext = createContext<TranslationContextType>(defaultState);
TranslationContext.displayName = "Translations Context";

function HtmlLangSync(props: { language: string }): React.ReactElement | null {
    const { language } = props;
    useEffect(() => {
        if (typeof document !== "undefined" && language) {
            document.documentElement.lang = language;
        }
    }, [language]);
    return null;
}

export function TranslationProvider({
    children,
    testValues
}: {
    children: ReactNode;
    testValues?: Record<string, string>;
}): React.ReactElement {
    const initialState = useMemo(() => getInitialState(), []);

    const [translations, setTranslations] = useState<Record<string, string>>(
        initialState.translations
    );
    const [basicTranslations, setBasicTranslations] = useState<
        Record<string, string>
    >(initialState.basicTranslations);
    const [country, setCountry] = useState<string>(initialState.country);
    const [language, setLanguage] = useState<string>(initialState.language);

    useEffect(() => {
        setCookie("country", country);
        setCookie("language", language);
        sendEvent("page_configration", {
            country,
            language
        });
    }, [country, language]);

    const fetchTranslations = debounce(
        async (_country: string, _language: string) => {
            try {
                const response = await translationsSvc.get({
                    country: _country || country,
                    language: _language || language
                });

                if (response.ok === false) {
                    throw new Error("Failed to fetch translations");
                }
                // const translations = await response.json();
                setTranslations(response);
            } catch (error) {
                console.error("Error fetching translations:", error);
            }
        },
        500
    );

    const updateCachedTranslationData = useCallback(() => {
        if (typeof window !== "undefined") {
            const dataToCache = {
                translations,
                basicTranslations,
                country,
                language
            };
            try {
                sessionStorage.setItem(
                    "translationContext",
                    JSON.stringify(dataToCache)
                );
            } catch (error) {
                console.error(
                    "Failed to save translation context data to sessionStorage:",
                    error
                );
            }
        }
    }, [translations, basicTranslations, country, language]);

    useEffect(() => {
        updateCachedTranslationData();
    }, [updateCachedTranslationData]);

    const value = useMemo(
        () => ({
            translations,
            basicTranslations,
            country,
            language,
            setTranslations,
            setBasicTranslations,
            setCountry,
            setLanguage,
            fetchTranslations
        }),
        [translations, basicTranslations, country, language, fetchTranslations]
    );

    const providerValue = useMemo(
        () => ({
            ...value,
            ...testValues
        }),
        [value, testValues]
    );

    return (
        <TranslationContext.Provider value={providerValue}>
            <HtmlLangSync language={language} />
            {children}
        </TranslationContext.Provider>
    );
}

export const useTranslations = (): TranslationContextType =>
    useContext(TranslationContext);
