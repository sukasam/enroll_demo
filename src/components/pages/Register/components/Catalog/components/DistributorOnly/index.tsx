/** @jsxImportSource @emotion/react */
import CurrencyFormatter from "Components/shared/CurrencyFormatter";
import { useTranslate } from "Components/shared/Translate";
import { useProducts } from "Contexts/ProductContext";
import { useUser } from "Contexts/UserContext";
import mixpanelService from "Services/mixpanel/initializeMixPanel";
import MixpanelEvent from "Services/mixpanel/mixpanelEvent";
import styles from "./styles";

function DistributorOnly(): JSX.Element {
    const { goToNextSection } = useUser();
    const translate = useTranslate();
    const { setShoppingCart, noPurchaseVariantData } = useProducts();
    return (
        <div css={styles}>
            <h3>{translate("catalog_distributor_only")}</h3>
            <div>
                <CurrencyFormatter>{0}</CurrencyFormatter> | PV: 0 |{" "}
                <span>{translate("catalog_payment_distributor_only")}</span>
            </div>
            <p>{translate("catalog_distributor_only_description")}</p>
            <button
                type="button"
                onClick={(): void => {
                    mixpanelService.trackEvent(
                        MixpanelEvent.REGISTER_STARTER_KIT_SELECTED,
                        {
                            basic_kit_selected: true,
                            basic_kit_available: true
                        }
                    );
                    setShoppingCart(noPurchaseVariantData);
                    goToNextSection();
                }}
                className="continue-button"
            >
                {translate("button_distributor_only")}
            </button>
        </div>
    );
}

export default DistributorOnly;
