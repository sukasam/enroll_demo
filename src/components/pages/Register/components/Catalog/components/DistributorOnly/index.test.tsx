import { render } from "@testing-library/react";
import { TranslationContextType } from "Contexts/translation";
import React from "react";
import DistributorOnly from "./index";

// Mock the necessary dependencies
jest.mock(
    "Components/shared/CurrencyFormatter",
    () =>
        function MockCurrencyFormatter({
            children
        }: {
            children: React.ReactNode;
        }): JSX.Element {
            return <span>{children}</span>;
        }
);

jest.mock("Contexts/UserContext", () => ({
    ...jest.requireActual("Contexts/UserContext"),
    useUser: (): { goToNextSection: jest.Mock } => ({
        goToNextSection: jest.fn()
    }),
    UserProvider: ({ children }: { children: any }): JSX.Element => children
}));

// Mock the translation context
const mockTranslationContext: TranslationContextType = {
    translations: {
        catalog_distributor_only: "Distributor Only",
        catalog_payment_distributor_only: "Payment Distributor Only",
        catalog_distributor_only_description: "Description",
        button_distributor_only: "Continue"
    },
    basicTranslations: {},
    country: "US",
    language: "en",
    setTranslations: jest.fn(),
    setBasicTranslations: jest.fn(),
    setCountry: jest.fn(),
    setLanguage: jest.fn(),
    fetchTranslations: jest.fn()
};

jest.mock("Contexts/translation", () => ({
    useTranslations: (): TranslationContextType => mockTranslationContext,
    TranslationProvider: ({ children }: { children: React.ReactNode }): any =>
        children
}));

describe("DistributorOnly", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<DistributorOnly />);
        expect(asFragment()).toMatchSnapshot();
    });
});
