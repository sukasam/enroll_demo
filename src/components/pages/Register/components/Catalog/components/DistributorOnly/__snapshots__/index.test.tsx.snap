// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistributorOnly renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-xkzasc"
  >
    <h3>
      Distributor Only
    </h3>
    <div>
      <span>
        0
      </span>
       | PV: 0 | 
      <span>
        Payment Distributor Only
      </span>
    </div>
    <p>
      Description
    </p>
    <button
      class="continue-button"
      type="button"
    >
      Continue
    </button>
  </div>
</DocumentFragment>
`;
